### DISCUSSION TITLE
Need help building ZMK firmware locally for custom Miryoku layout

### DISCUSSION THREAD

**Discussion by MilesHoover** [manna-harbour/miryoku#299] [Q&A] [answered]
For full context, I am attempting to build ZMK firmware locally using the Miryoku layout on my wireless Corne 3x6.I've successfully built firmware when the custom_config.h is blank so the issue is certainly there but I cannot figure this out for the life of me.

[Here are the build fail logs](https://github.com/MilesHoover/miryoku-zmk/blob/master/build-fail-logs.txt)

- the biggest takeaway from these logs is the:'devicetree error: /home/miles/Documents/zmk-build-env/miryoku-zmk/config/../miryoku/miryoku.dtsi:23 (column 1499): parse error: expected number or parenthesized expression'

[Here is my custom_config.h file](https://github.com/MilesHoover/miryoku-zmk/blob/master/miryoku/custom_config.h)

- Like I mentioned previously, I'm certainly doing something wrong here but I'm just not sure what it is and need a second pair of eyes.

- I am following the instructions for mapping custom keys using whats written in the [Miryoku Customization Documentation](https://github.com/manna-harbour/miryoku/discussions/85)

**Comment by manna-harbour** [comment]
You're missing `&kp` before `RGUI` here https://github.com/MilesHoover/miryoku-zmk/blob/c6cae12c1100aee3c7ad7a6093063ad67729a5d8/miryoku/custom_config.h#L14.

**Reply by MilesHoover** [reply]
Wow, I never would have seen that. I fixed that and pushed the update. 

I am still getting the exact same error, however. I updated the build-fail-logs.txt file.

Is my '#define MIRYOKU_ALPHAS_COLEMAKDH' in the right place at the bottom of the file or does it need to be in the main block?

Thanks so much

**Reply by manna-harbour** [reply]
You should only have this section https://github.com/MilesHoover/miryoku-zmk/blob/13c2969e9c6e94a4b7039b35d4c7ed3c0ff90391/miryoku/custom_config.h#L6-L30, not the surrounding parts.

`#define MIRYOKU_ALPHAS_COLEMAKDH` looks ok, but it's the default so it's not needed.

**Reply by manna-harbour** [reply, marked as answer]
`RCRTL` should be `RCTRL`.

**Reply by manna-harbour** [reply]
`parse error: expected number or parenthesized expression` would generally indicate an incorrect keycode.
